### PR TITLE
Adds support for target

### LINKED ISSUE #45: Missing `this.target` for loaders
Currently `target` is missing and is used by one of the loaders I'm using

### PR DESCRIPTION
**Pull request by Akkuma**
Fixes #45 by adding in `target`. I also "cleaned up" `HappyFakeLoaderContext`, while I was in there.

### CODE REVIEW

File: lib/HappyFakeLoaderContext.js [hunk 1]
```diff
@@ -1,91 +1,97 @@
-function HappyFakeLoaderContext(initialValues) {
-  var loader = {};
```
Comment (amireh) [inline lib/HappyFakeLoaderContext.js]: Hmm, I'm having a hard time understanding why this was necessary?

**Comment by amireh** [comment]
Thanks for the help! I'm just a little unsure about the "clean up" change as you put it, why is it necessary? How is the new style any better?

**Comment by Akkuma** [comment]
The new style to me is better as to not repeat loader over and over, and there are possible performance implications of creating objects with no properties set on them. I could optimize this more, but for what I was trying to do that went another step beyond the scope.

A few different sources about this:
http://ariya.ofilabs.com/2012/02/javascript-object-structure-speed-matters.html
http://mrale.ph/blog/2014/07/30/constructor-vs-objectcreate.html
http://mrale.ph/blog/2013/08/14/hidden-classes-vs-jsperf.html
https://developers.google.com/v8/design#fast-property-access

**Comment by amireh** [comment]
Thanks for sharing - I am aware of the performance implications. It was prototypal at first (https://github.com/amireh/happypack/commit/b3086ecf98cfc3d7752f02123ee5385e782205cf#diff-6b7922d9a93503a44461e1218952e04fL1), until I realized that webpack doesn't do it and loaders keep references to context methods without binding (e.g. `var x = this.callback` instead of `var x = this.callback.bind(this)`) which doesn't work obviously.

Regarding the style itself, I do disagree, which isn't a big deal normally since people hardly agree on styles but in this case it messes up the blame stats and I don't want to lose those stats so if you don't mind let's keep it the way it was.

**Comment by amireh** [comment]
Thank you! Looks great.

**Pull request merged**